File: backend/expenses-service/src/main/java/com/ironhack/expensesservice/model/Expense.java
package com.ironhack.expensesservice.model;

import javax.persistence.*;
import java.math.BigDecimal;
import java.util.Date;

@Entity
public class Expense {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private BigDecimal amount;
    private String description;
    @ManyToOne
    @JoinColumn(name = "users_id")
    private Users paidBy;
    @ManyToOne
    @JoinColumn(name = "party_id")
    private Party party;

    private Date creationDate;
//    @OneToMany(mappedBy = "expense")
//    private List<Split> splits;

    public Expense(BigDecimal amount, String description, Users paidBy, Party party) {
        this.amount = amount;
        this.description = description;
        this.paidBy = paidBy;
        this.party = party;
        this.creationDate = new Date();
    }

    public Expense() {
        this.creationDate = new Date();
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public BigDecimal getAmount() {
        return amount;
    }

    public void setAmount(BigDecimal amount) {
        this.amount = amount;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Users getPaidBy() {
        return paidBy;
    }

    public void setPaidBy(Users paidBy) {
        this.paidBy = paidBy;
    }

    public Date getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(Date creationDate) {
        this.creationDate = creationDate;
    }

    public Party getParty() {
        return party;
    }

    public void setParty(Party party) {
        this.party = party;
    }

    //    public List<Split> getSplits() {
//        return splits;
//    }
//
//    public void setSplits(List<Split> splits) {
//        this.splits = splits;
//    }
}

File: frontend/src/app/model/expense.ts
import { User } from "./user";

export class Expense {
    constructor(
        private _id: number, 
        private _amount: number, 
        private _description: string, 
        private _paidBy: User,
        private _partyId: number, 
        private _creationDate: Date
    ){}

    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }

    public get amount(): number {
        return this._amount;
    }
    public set amount(value: number) {
        this._amount = value;
    }

    public get description(): string {
        return this._description;
    }
    public set description(value: string) {
        this._description = value;
    }

    public get paidBy(): User {
        return this._paidBy;
    }
    public set paidBy(value: User) {
        this._paidBy = value;
    }

    public get partyId(): number {
        return this._partyId;
    }
    public set partyId(value: number) {
        this._partyId = value;
    }

    public get creationDate(): Date {
        return this._creationDate;
    }
    public set creationDate(value: Date) {
        this._creationDate = value;
    }

}
